qiskit/ignis/experiments/calibration/instruction_data/gate_and_schedules_definition.py

```diff
@@ -55,243 +78,196 @@ def from_backend(cls, backend: BaseBackend) -> 'InstructionsDefinition':
+    @property
+    def pulse_channels(self) -> Enum:
+        """Return the definition of pulse channels."""
+        return self._pulse_channels
 
-        return InstructionsDefinition(
-            backend_name=backend.name(),
-            n_qubits=backend.configuration().n_qubits,
-            channel_qubit_map=channel_qubit_map,
-            pulse_table=pulse_table)
+    @property
+    def series(self) -> str:
+        """Return the current data series to construct gate schedule."""
+        return self._series
+
+    @series.setter
+    def series(self, new_series):
+        """Set new series name."""
+        self._series = new_series
 
     @property
-    def instructions(self) -> Dict[Tuple[str, Iterable], Schedule]:
-        """Return the instructions as a dict."""
+    def instructions(self) -> pd.DataFrame:
+        """Return the instructions as a data frame."""
         return self._instructions
 
     @property
-    def pulse_parameter_table(self) -> PulseTable:
+    def pulse_parameter_table(self) -> PulseParameterTable:
         """Returns the table of pulse parameters."""
         return self._pulse_table
 
     def get_calibration(self):
         """Returns the calibrations."""
         return self._pulse_table.filter_data()
 
-    def get_circuit(self, name: str, qubits: Tuple,
+    def get_circuit(self,
+                    gate_name: str,
+                    qubits: Tuple,
                     free_parameter_names: List[str] = None) -> QuantumCircuit:
         """
         Wraps the schedule from the instructions table in a quantum circuit.
 
         Args:
-            name: name of the instruction to retrive
+            gate_name: name of the instruction to retrive
             qubits: qubits to which the instruction applies.
             free_parameter_names: Names of the parameter that should be left unbound.
                 If None is specified then all parameters will be bound to their
                 calibrated values.
         """
-        schedule = self.get_schedule(name, qubits, free_parameter_names)
+        schedule = self.get_gate_schedule(gate_name, qubits, free_parameter_names)
 
-        gate = Gate(name=name, num_qubits=len(qubits), params=schedule.parameters)
+        gate = Gate(name=gate_name, num_qubits=len(qubits), params=list(schedule.parameters))
         circ = QuantumCircuit(self._n_qubits)  # Probably a better way of doing this
         circ.append(gate, qubits)
         circ.add_calibration(gate, qubits, schedule, params=schedule.parameters)
 
         return circ
 
-    def get_schedule(self, name: str, qubits: Tuple,
-                     free_parameter_names: Optional[List[str]] = None) -> Schedule:
+    def add_gate_schedule(self,
+                          gate_name: str,
+                          qubits: Union[int, Tuple[int]],
+                          signature: List[str],
+                          schedule: Schedule):
+        """"""
```

Comment:
What is the intention behind this?

Reply:
I just think what we have in the inst def is schedule for gate, because we changed the generator model (to generate a gate with attached schedule). But I'm ok with using `get_schedule`.